modules/ocl/cl_image_360_stitch.cpp

```diff
@@ -806,21 +810,86 @@ CLImage360Stitch::sub_handler_execute_done (SmartPtr<CLImageHandler> &handler)
+        config.min_corners = 4;
+        config.offset_factor = 0.8f;
+        config.delta_mean_offset = 120.0f;
+        config.recur_offset_error = 8.0f;
+        config.max_adjusted_offset = 24.0f;
+        config.max_valid_offset_y = 20.0f;
+        config.max_track_error = 28.0f;
+
+        if (handler.ptr () == _fisheye[_fisheye_num - 1].handler.ptr ()) {
+            int idx_next = 1;
+            Rect crop_left, crop_right;
+            Rect tmp_crop_left, tmp_crop_right;
+
+            for (int i = 0; i < _fisheye_num; i++) {
+                idx_next = (i == (_fisheye_num - 1)) ? 0 : (i + 1);
+
+                convert_to_stitch_rect (_img_merge_info[i].right, crop_left);
+                convert_to_stitch_rect (_img_merge_info[idx_next].left, crop_right);
+
+                tmp_crop_left = crop_left;
+                tmp_crop_right = crop_right;
+
+                _feature_match[i]->set_config (config);
+                _feature_match[i]->reset_offsets ();
+                _feature_match[i]->optical_flow_feature_match (
+                    _fisheye[i].buf, _fisheye[idx_next].buf, tmp_crop_left, tmp_crop_right, _fisheye[i].width);
+
+                float left_offsetx = _feature_match[i]->get_current_left_offset_x ();
+                PointFloat2 left_factor, right_factor;
+                uint32_t left_idx = i;
+                float center_x = (float) _fisheye[left_idx].width / 2;
+                float feature_center_x = (float)crop_left.pos_x + crop_left.width / 2.0f;
+                float range = feature_center_x - center_x;
+                XCAM_ASSERT (range > 1.0f);
+                right_factor.x = (range + left_offsetx / 2.0f) / range;
+                right_factor.y = 1.0f;
+                XCAM_ASSERT (right_factor.x > 0.0f && right_factor.x < 2.0f);
+
+                uint32_t right_idx = (i + 1) % _fisheye_num;
+                center_x = (float)_fisheye[right_idx].width / 2;
+                feature_center_x = (float)crop_right.pos_x + crop_right.width / 2.0f;
+                range = center_x - feature_center_x;
+                XCAM_ASSERT (range > 1.0f);
+                left_factor.x = (range + left_offsetx / 2.0f) / range;
+                left_factor.y = 1.0f;
+                XCAM_ASSERT (left_factor.x > 0.0f && left_factor.x < 2.0f);
+
+                PointFloat2 last_left_factor, last_right_factor;
+                last_left_factor = _fisheye[right_idx].handler->get_left_scale_factor ();
+                last_right_factor = _fisheye[left_idx].handler->get_right_scale_factor ();
+
+                left_factor.x *= last_left_factor.x;
+                left_factor.y *= last_left_factor.y;
+                right_factor.x *= last_right_factor.x;
+                right_factor.y *= last_right_factor.y;
+
+                _fisheye[left_idx].handler->set_right_scale_factor (right_factor);
+                _fisheye[right_idx].handler->set_left_scale_factor (left_factor);
+            }
         }
```

Comment:
this function is too long. please use simple ways to implement. otherwise, the code may become very hard to maintain.
1. CVFMConfig config; should be set only the 1st time, e.g. in configure function.
2.  the 2 branch(if/else) still have lots same code.  consider to reduce. 

		if (handler.ptr () == _fisheye[_fisheye_num - 1].handler.ptr ()) {
			int idx_next = 1;
			Rect crop_left, crop_right;
			Rect tmp_crop_left, tmp_crop_right;

			for (int i = 0; i < _fisheye_num; i++) {
				idx_next = (i == (_fisheye_num - 1)) ? 0 : (i + 1);

				convert_to_stitch_rect (_img_merge_info[i].right, crop_left);
				convert_to_stitch_rect (_img_merge_info[idx_next].left, crop_right);
				
				if (_surround_mode == SphereView) {
					...
				} else {
					...
					adjust_scale_factors ();
				}
			}